app/components/post_list/post_list.tsx

```diff
@@ -172,15 +178,31 @@ const PostList = ({
             map((post) => removePost(serverUrl, post));
         await Promise.all(removalPromises);
         setRefreshing(false);
-    }, [channelId, location, posts, rootId, disablePullToRefresh]);
+    }, [disablePullToRefresh, location, channelId, rootId, posts, serverUrl]);
+
+    const scrollToIndex = useCallback((index: number, animated = true, applyOffset = true) => {
+        listRef.current?.scrollToIndex({
+            animated,
+            index,
+            viewOffset: applyOffset ? Platform.select({ios: -45, default: 0}) : 0,
+            viewPosition: 1, // 0 is at bottom
+        });
+    }, []);
 
     const onScroll = useCallback((event: NativeSyntheticEvent<NativeScrollEvent>) => {
         const {y} = event.nativeEvent.contentOffset;
-        if (y === 0) {
-            setLastPostId(posts[0]?.id);
+        const isThresholdReached = y > CONTENT_OFFSET_THRESHOLD;
+
+        // console.log({y, isThresholdReached, showScrollToEndBtn, lastPostId, firstIdInPosts});
+
+        if (isThresholdReached !== showScrollToEndBtn) {
+            setShowScrollToEndBtn(isThresholdReached);
         }
-        setShowScrollToEndBtn(y > CONTENT_OFFSET_THRESHOLD);
-    }, [posts[0]?.id]);
+
+        if (!y && lastPostId !== firstIdInPosts) {
```

Comment:
This should also apply if the post owner is not the current user imho

Reply:
@enahum  Should I change that?  I feel like I would deviate from the actual fix and going out of scope for something that could be experienced differently?

Reply:
Well in my mind makes sense to change🤪, but we can open a ticket to be triaged and closed if this is the behavior we want when the current user post messages

Reply:
@enahum I've added a ticket for this: https://mattermost.atlassian.net/browse/MM-62821.

Can you unblock me so I can merge this PR?

Or is there another thing you think I should look into vs the scrollEventThrottle change that I made?

Reply:
Approved although it would be nice to see the diff in fps while scrolling eith the previous and new value on the throttle

Reply:
Oh sorry, I should do that first.